File: tools/apply-character-schema.js
#!/usr/bin/env node

/**
 * Apply Character System Schema
 * Creates all character-related tables in the database
 */

import pg from 'pg';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const { Pool } = pg;

// Use hardcoded values from MCP config since .env loading is problematic
const pool = new Pool({
  host: process.env.POSTGRES_HOST || '127.0.0.1',
  port: parseInt(process.env.POSTGRES_PORT || '5434'),
  user: process.env.POSTGRES_USER || 'postgres',
  password: process.env.POSTGRES_PASSWORD || 'daf562f1a5bccb27ff6a7de5c0b5Wq',
  database: process.env.POSTGRES_DB || 'postgres',
});

async function applySchema() {
  console.log('Applying character system schema...\n');

  try {
    // Read the schema file
    const schemaPath = path.join(__dirname, '..', 'schema', 'character_system.sql');
    const schema = fs.readFileSync(schemaPath, 'utf8');

    // Execute the schema
    await pool.query(schema);

    console.log('✓ Character system schema applied successfully!');
    console.log('\nTables created:');
    console.log('  - campaigns');
    console.log('  - metatypes');
    console.log('  - sr_characters');
    console.log('  - character_skills');
    console.log('  - character_spells');
    console.log('  - character_powers');
    console.log('  - character_gear');
    console.log('  - character_contacts');
    console.log('  - character_history');
    console.log('  - character_modifiers');

    // Insert default metatypes
    console.log('\nInserting default metatypes...');
    await pool.query(`
      INSERT INTO metatypes (name, body_range, quickness_range, strength_range, charisma_range, intelligence_range, willpower_range, reaction_range, essence_range, magic_range)
      VALUES 
        ('human', ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 6], ARRAY[0, 6]),
        ('elf', ARRAY[1, 6], ARRAY[1, 7], ARRAY[1, 6], ARRAY[1, 8], ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 6], ARRAY[0, 6]),
        ('dwarf', ARRAY[1, 7], ARRAY[1, 6], ARRAY[1, 7], ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 7], ARRAY[1, 6], ARRAY[1, 6], ARRAY[0, 6]),
        ('ork', ARRAY[1, 8], ARRAY[1, 6], ARRAY[1, 8], ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 6], ARRAY[0, 6]),
        ('troll', ARRAY[1, 9], ARRAY[1, 6], ARRAY[1, 9], ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 6], ARRAY[1, 6], ARRAY[0, 6])
      ON CONFLICT (name) DO NOTHING
    `);

    console.log('✓ Default metatypes inserted!');
    console.log('\nSchema application complete!');
  } catch (error) {
    console.error('Error applying schema:', error.message);
    console.error(error.stack);
    process.exit(1);
  } finally {
    await pool.end();
  }
}

applySchema();

File: tools/apply-character-schema.py
#!/usr/bin/env python3
"""
Apply character system schema to database
"""
import psycopg2
import os
from dotenv import load_dotenv
from pathlib import Path

load_dotenv()

# Database configuration
db_config = {
    'host': os.getenv('POSTGRES_HOST', 'localhost'),
    'port': int(os.getenv('POSTGRES_PORT', 5432)),
    'database': os.getenv('POSTGRES_DB', 'postgres'),
    'user': os.getenv('POSTGRES_USER', 'postgres'),
    'password': os.getenv('POSTGRES_PASSWORD', '')
}

# Read schema file
schema_file = Path(__file__).parent.parent / 'schema' / 'character_system.sql'
with open(schema_file, 'r', encoding='utf-8') as f:
    schema_sql = f.read()

# Connect and apply schema
print("Connecting to database...")
conn = psycopg2.connect(**db_config)
conn.autocommit = True

try:
    print("Applying character system schema...")
    cur = conn.cursor()
    cur.execute(schema_sql)
    print("✓ Schema applied successfully!")
    
    # Check what was created
    cur.execute("""
        SELECT table_name 
        FROM information_schema.tables 
        WHERE table_schema = 'public' 
        AND table_name IN (
            'metatypes', 'qualities', 'skills', 'powers', 'spells', 'totems',
            'creatures', 'characters', 'contacts', 
            'reference_load_history', 'character_history'
        )
        ORDER BY table_name
    """)
    
    tables = cur.fetchall()
    print(f"\n✓ Created {len(tables)} tables:")
    for table in tables:
        print(f"  - {table[0]}")
    
    # Check views
    cur.execute("""
        SELECT table_name 
        FROM information_schema.views 
        WHERE table_schema = 'public' 
        AND table_name IN ('character_summary', 'reference_data_stats')
        ORDER BY table_name
    """)
    
    views = cur.fetchall()
    if views:
        print(f"\n✓ Created {len(views)} views:")
        for view in views:
            print(f"  - {view[0]}")
    
    # Show reference data stats
    cur.execute("SELECT * FROM reference_data_stats ORDER BY table_name")
    stats = cur.fetchall()
    print("\n📊 Current data counts:")
    for stat in stats:
        print(f"  {stat[0]:20} {stat[1]:4} items")
    
finally:
    conn.close()
    print("\n✓ Done!")
